lisa/tools/lsblk.py

```diff
@@ -228,3 +234,109 @@ def find_mountpoint_by_volume_name(
+
+    # Example:
+    # da1
+    _DISK_NAME_REGEX_MATCH = re.compile(r"^\D+\d*$")
+
+    # Example:
+    # da1p1
+    _PARTITION_NAME_REGEX_MATCH = re.compile(r"^\D+\d+\D*\d+$")
+
+    _PARTITION_DISK_NAME_REGEX = re.compile(r"^(?P<disk_name>\D+\d+)\D+\d+$")
+
+    @property
+    def command(self) -> str:
+        return "lsblk"
+
+    @property
+    def can_install(self) -> bool:
+        return True
+
+    def _install(self) -> bool:
+        assert isinstance(self.node.os, BSD)
+        self.node.os.install_packages("lsblk")
+        return self._check_exists()
+
+    def get_disks(self, force_run: bool = False) -> List[DiskInfo]:
+        disks: List[DiskInfo] = []
+
+        # parse output of lsblk
+        output = self.run().stdout
+        entries = find_patterns_groups_in_lines(output, [self._ENTRY_REGEX])[0]
+
+        # create partition map to store partitions for each disk
+        disk_partition_map: Dict[str, List[PartitionInfo]] = {}
+
+        # get partitions for each disk
+        for entry in entries:
+            entry_name = entry["name"]
+            if not self._PARTITION_NAME_REGEX_MATCH.match(entry_name):
+                continue
+
+            # extract drive name from partition name
+            matched = find_patterns_groups_in_lines(
+                entry["name"], [self._PARTITION_DISK_NAME_REGEX]
+            )[0]
+            assert len(matched) == 1, "Could not extract drive name from partition name"
+            drive_name = matched[0]["disk_name"]
+
+            # convert size to bytes
+            size_in_bytes = self._get_size_in_bytes(
+                int(entry["size"]), entry["size_unit"]
+            )
+
+            # add partition to disk partition map
+            if drive_name not in disk_partition_map:
+                disk_partition_map[drive_name] = []
+            disk_partition_map[drive_name].append(
+                PartitionInfo(
+                    name=entry["name"],
+                    size=size_in_bytes,
+                    mountpoint=entry["mountpoint"],
+                    fstype=entry["type"],
+                )
+            )
+
+        # get disk info
+        for entry in entries:
+            entry_name = entry["name"]
+            if not self._DISK_NAME_REGEX_MATCH.match(entry_name):
+                continue
+
+            # convert size to bytes and create disk info
+            size_in_bytes = self._get_size_in_bytes(
+                int(entry["size"]), entry["size_unit"]
+            )
+            disks.append(
+                DiskInfo(
+                    name=entry["name"],
+                    mountpoint=entry["mountpoint"],
+                    size=size_in_bytes,
+                    dev_type=entry["type"],
+                    partitions=disk_partition_map.get(entry["name"], []),
+                )
+            )
+
+        # sort disk with OS disk first
+        disks.sort(key=lambda disk: disk.is_os_disk, reverse=True)
+        return disks
+
+    @staticmethod
```

Comment:
This method doesn't need to be static.

Reply:
We are not using any class information. We can also move it to common function if you feel so

Reply:
You can move it to a function in the same file. It's not shared by other classes, so it doesn't need to move to common.